File: Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int size = A.size();
        vector<int> B(size);
        int evenIdx = 0, i = 0, oddIdx = size-1;
        while(evenIdx <= oddIdx){
            if(A[i]%2==0){
                B[evenIdx] = A[i];
                evenIdx++;
                
            }else{
                B[oddIdx] = A[i];
                oddIdx--;
            }
            i++;
        }
        return B;
    }
};

File: Sort Array By Parity.py
class Solution(object):
    def sortArrayByParity(self, A):
        size = len(A)
        arr = [-1]*size
        evenIdx = 0
        oddIdx = size-1
        i = 0
        while evenIdx <= oddIdx:
            if A[i] % 2 == 0:
                arr[evenIdx] = A[i]
                evenIdx+=1
            else:
                arr[oddIdx] = A[i]
                oddIdx-=1
            i+=1
        return arr
        
            

        
        
